### DISCUSSION TITLE
API Reference Tab

### DISCUSSION THREAD

**Discussion by eggorybarnes** [teambit/bit#8710] [General]
Hey,

I'm just exploring the platform and was hoping someone could help clarify something for me. I was wondering how the API Reference tab works? Does it automatically pull from the code? Do you need to add comments for it to work? What about routes on an api can it create documentation from those? 

Thanks in advance
Greg

**Comment by itaymendel** [comment]
Bit runs an AST-schema-extraction using TypeScript and uses it to build the API-reference. So it knows all things you export and their types. It is then uses this information to document the API-ref.
You can use all JSDocs annotations, and Bit captures it to add descriptions, examples and more.

This is more about how you programmatic APIs are, for developers who are looking to integrate with your component.
